packages/components/date-picker/src/date-picker-com/basic-year-table.vue

```diff
@@ -59,54 +67,181 @@ const startYear = computed(() => {
+        cell.start = !!(calEndDate && calTime.isSame(calEndDate, 'year'))
+        cell.end = !!(props.minDate && calTime.isSame(props.minDate, 'year'))
+      } else {
+        cell.start = !!(props.minDate && calTime.isSame(props.minDate, 'year'))
+        cell.end = !!(calEndDate && calTime.isSame(calEndDate, 'year'))
+      }
+
+      const isToday = now.isSame(calTime)
+      if (isToday) {
+        cell.type = 'today'
+      }
+      cell.text = index
+      const cellDate = calTime.toDate()
+      cell.disabled =
+        (props.disabledDate && props.disabledDate(cellDate)) || false
+      row[j] = cell
+    }
+  }
+
+  return rows
+})
+
 const focus = () => {
   currentCellRef.value?.focus()
 }
 
-const getCellKls = (year: number) => {
+const getCellKls = (cell: YearCell) => {
   const kls: Record<string, boolean> = {}
   const today = dayjs().locale(lang.value)
+  const year = cell.text
 
   kls.disabled = props.disabledDate
     ? datesInYear(year, lang.value).every(props.disabledDate)
     : false
 
+  kls.today = today.year() === year
   kls.current =
     castArray(props.parsedValue).findIndex((d) => d!.year() === year) >= 0
 
-  kls.today = today.year() === year
+  if (cell.inRange) {
+    kls['in-range'] = true
 
+    if (cell.start) {
+      kls['start-date'] = true
+    }
+
+    if (cell.end) {
+      kls['end-date'] = true
+    }
+  }
   return kls
 }
 
-const isSelectedCell = (year: number) => {
-  return (
-    (year === startYear.value &&
-      props.date.year() < startYear.value &&
-      props.date.year() > startYear.value + 9) ||
-    castArray(props.date).findIndex((date) => date.year() === year) >= 0 ||
-    castArray(props.parsedValue).findIndex((date) => date?.year() === year) >= 0
-  )
+const isSelectedCell = (cell: YearCell) => {
+  const year = cell.text
+  return castArray(props.date).findIndex((date) => date.year() === year) >= 0
 }
 
 const handleYearTableClick = (event: MouseEvent | KeyboardEvent) => {
-  const clickTarget = event.target as HTMLDivElement
-  const target = clickTarget.closest('td')
-  if (target && target.textContent) {
-    if (hasClass(target, 'disabled')) return
-    const year = target.textContent || target.innerText
-    if (props.selectionMode === 'years') {
-      if (event.type === 'keydown') {
-        emit('pick', castArray(props.parsedValue), false)
-        return
-      }
-      const newValue = hasClass(target, 'current')
-        ? castArray(props.parsedValue).filter((d) => d?.year() !== Number(year))
-        : castArray(props.parsedValue).concat([dayjs(year)])
-      emit('pick', newValue)
+  const target = (event.target as HTMLElement)?.closest(
+    'td'
+  ) as HTMLTableCellElement
+  if (!target || !target.textContent || hasClass(target, 'disabled')) return
+
+  if (props.selectionMode === 'range') {
+    const column = target.cellIndex
+    const row = (target.parentNode as HTMLTableRowElement).rowIndex
+
+    const selectedYear = row * 4 + column + startYear.value
```

Comment:
To get the year value of the currently clicked cell, it should be possible to directly use the original method to get the text content of the `td` tag. Are there any considerations for using the current process?

Reply:
IMO, this calculation will be more accurate,
First, dom can be changed manually, and if it is changed, the wrong value will be obtained. Second, if td needs to be written in a non-numeric form in some possible new feature, such as 二零二零, it will also be problematic to obtain td directly (although this requirement is unlikely to occur).
What do you think?

Reply:
Make sense.